src/SparqlHandler.js

```diff
@@ -1,37 +1,116 @@
 /**
- * Expresses a path as a SPARQL query.
+ * Expresses a path or mutation as a SPARQL query.
  *
  * Requires:
- * - a pathExpression property on the path proxy
+ * - a mutationExpressions or pathExpression property on the path proxy
  */
 export default class SparqlHandler {
   async execute(path, proxy) {
+    // First check if we have a mutation expression
+    const mutationExpressions = await proxy.mutationExpressions;
+    if (Array.isArray(mutationExpressions) && mutationExpressions.length)
+      return this.executeMutationExpression(path, proxy, mutationExpressions);
+
+    // Otherwise, fallback to checking for a path expression
     const pathExpression = await proxy.pathExpression;
     if (!Array.isArray(pathExpression))
       throw new Error(`${path} has no pathExpression property`);
+    return this.executePathExpression(path, proxy, pathExpression);
+  }
 
+  executePathExpression(path, proxy, pathExpression) {
     // Require at least a subject and a link
     if (pathExpression.length < 2)
       throw new Error(`${path} should at least contain a subject and a predicate`);
-    const root = pathExpression.shift();
 
     // Determine the query variable name
     const queryVar = path.property.match(/[a-z0-9]*$/i)[0] || 'result';
 
     // Build basic graph pattern
-    const last = pathExpression.length - 1;
+    const clauses = this.expressionToTriplePatterns(pathExpression, queryVar);
+
+    // Embed the basic graph pattern into a SPARQL query
+    const joinedClauses = clauses.join('\n  ');
+    return `SELECT ?${queryVar} WHERE {\n  ${joinedClauses}\n}`;
+  }
+
+  executeMutationExpression(path, proxy, mutationExpressions) {
+    return mutationExpressions
+      .map(mutationExpression => this.mutationExpressionToQuery(mutationExpression))
+      .join('\n;\n');
+  }
+
+  expressionToTriplePatterns(pathExpression, queryVar, variableScope = {}) {
+    const root = pathExpression[0];
+    const last = pathExpression.length - 2;
     let object = `<${root.subject}>`;
-    const clauses = pathExpression.map((segment, index) => {
+    return pathExpression.slice(1).map((segment, index) => {
```

Comment:
I don't really like the number arithmetic above. Is it clearer to use array destructuring?